spec/02-integration/05-proxy/10-balancer/01-healthchecks_spec.lua

```diff
@@ -899,9 +927,9 @@ for _, strategy in helpers.each_strategy() do
                   name = upstreams[2].name,
                 })
 
-                if consistency == "eventual" then
+                --if consistency == "eventual" then
                   ngx.sleep(bu.CONSISTENCY_FREQ) -- wait for proxy state consistency timer
-                end
+                --end
```

Comment:
why commented out?

Reply:
With then new event mechanism, we always have 'eventual consistency' behavior.

Reply:
> we always have 'eventual consistency' behavior

I don't understand it.

Does this mean that all events are passed to all workers in an instant?

Reply:
https://github.com/Kong/kong/blob/0973036503941c52c56f8a1be74b686a62a40cdd/kong/runloop/handler.lua#L1138-L1156

https://github.com/Kong/kong/blob/0973036503941c52c56f8a1be74b686a62a40cdd/kong/runloop/handler.lua#L850-L852

https://github.com/Kong/kong/blob/0973036503941c52c56f8a1be74b686a62a40cdd/kong/runloop/handler.lua#L560-L572

For example, in the case of router updates, there is always a timer to update it periodically, regardless of the `worker_consistency` policy.

If the consistency policy is "strict", the route will be rebuilt each time it is fetched (if necessary), and otherwise, it will only be rebuilt by the timer.

So I think if the consistency policy is `strict`, we should not call `ngx.sleep()`.

Reply:
With the new event mechanism, perhaps we will have no `strict`, the async event publish/subscribe mechanism can not guarantee the timing of the router's rebuild.

Reply:
@chronolaw `strict` should go away. I would say one of the only reasons it is there is that otherwise a lot of tests will start to fail. Kong in already very `eventual` with `cluster events` delays and `worker events` delays or in hybrid, the websockets delays, exporting delays, applying new config delays etc. Strict basically just means whether or not Kong refuses to e.g. use a stale router and request a new one on request time. I don't think it makes a much sense.

The `strict` vs `eventual` should just go away and everything should work the way it works `eventual` currently, aka things are build on background, and not on request time.

Reply:
Yes, but in this PR, I just keep the `strict` in comments, perhaps we should remove it at a later focus PR.

Reply:
https://github.com/Kong/kong/pull/8842